### PR TITLE
[MSFT] Add PhysicalRegion op and related attributes.

### PR DESCRIPTION
**Pull request by mikeurbach**
A PhysicalRegion is a Symbol which can be referred to by ops to be
placed within it. It contains potentially multiple rectangular bounds
on the physical device.

### CODE REVIEW

File: include/circt/Dialect/MSFT/MSFTAttributes.td [hunk 1]
```diff
@@ -61,3 +61,30 @@ def PhysLocation : MSFT_Attr<"PhysLocation"> {
     "PrimitiveTypeAttr":$primitiveType,
     "uint64_t":$x, "uint64_t":$y, "uint64_t":$num);
 }
+
+def PhysicalBounds : MSFT_Attr<"PhysicalBounds"> {
+  let summary = "Describes a rectangle bounding a physical region on a device";
```
Comment (teqdruid) [inline include/circt/Dialect/MSFT/MSFTAttributes.td:66]: Should probably specify whether or not it's inclusive in a description.

File: include/circt/Dialect/MSFT/MSFTAttributes.td [hunk 2]
```diff
@@ -61,3 +61,30 @@ def PhysLocation : MSFT_Attr<"PhysLocation"> {
...
+  let mnemonic = "physical_bounds";
+  let parameters = (ins
```
Comment (teqdruid) [inline include/circt/Dialect/MSFT/MSFTAttributes.td:68]: What should we do for "unbounded" boxes? Is the thing which creates this expected to know the maximum y,x and use that? That's a totally reasonable approach, IMO. If so, that should be discussed in the description.

**Review by teqdruid — APPROVED** [review]
LGTM

File: include/circt/Dialect/MSFT/MSFTAttributes.td (hunk 2 above)
Comment (mikeurbach) [reply include/circt/Dialect/MSFT/MSFTAttributes.td:68]: Yes my intention is whatever creates these has to specify lower and upper bounds in the x and y axes. I can document this.

File: include/circt/Dialect/MSFT/MSFTAttributes.td (hunk 1 above)
Comment (mikeurbach) [reply include/circt/Dialect/MSFT/MSFTAttributes.td:66]: Good point... I will look up which it is an mention that.

**Pull request merged**